src/Controller/Api/FormController.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace AppBundle\Controller\Api;
+
+use AppBundle\Form\AdherentRegistrationType;
+use AppBundle\Form\BecomeAdherentType;
+use AppBundle\Form\DonationSubscriptionRequestType;
+use AppBundle\Form\UserRegistrationType;
+use JMS\Serializer\Serializer;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * @Route("/form")
+ */
+class FormController extends Controller
+{
+    // TODO: Replace this by a compiler pass who crawl all FormTypeApiExposeInterface to add key and class in a FormTypeApiExposeManager
+    private const FORM_MAP = [
```

Comment:
After all you should not need this map. "internal" urls do not have to be pretty, and you should better use the FQCN of the types as a POST parameter.

Reply:
And no need for a compiler pass either ;).